catalog/CatalogDatabaseCache.cpp

```diff
@@ -61,15 +61,17 @@ void CatalogDatabaseCache::update(const serialization::CatalogDatabase &proto) {
 
   SpinSharedMutexExclusiveLock<false> lock(relations_mutex_);
   for (int i = 0; i < proto.relations_size(); ++i) {
-    auto relation_schema = make_unique<const CatalogRelationSchema>(proto.relations(i));
-    const relation_id rel_id = relation_schema->getID();
+    const serialization::CatalogRelationSchema &proto_relation = proto.relations(i);
+    const relation_id rel_id = proto_relation.relation_id();
```

Comment:
I am not sure how much performance intensive this function will be.  However, If performance is a concern we can refactor it as follows:

```
// No protection needed for this -
vector<rel_id> relation_ids;
// populate relation_ids from proto.

vector<relation_id> need_to_add_relation_id

// In a shared lock mode
for each rel_id in relation_ids {
  if (rel_id not found in rel_map_) {
    need_to_add_relation_id.empalce(rel_id)
  }
}
if (!need_to_add_relation_id.empty()) {
// Exclusive lock
// Add the necessary relation IDs to rel_map_.
}
```